src/v2/stack_aggregator.py

```diff
@@ -351,13 +358,168 @@ def execute(request: Dict, persist=True):
+            bindings['packages'] = list(packages)
+            started_at = time.time()
+            result = post_gremlin(query, bindings)
+            logger.info(
+                '%s took %0.2f secs for post_gremlin() batch request',
+                self._request.external_request_id, time.time() - started_at)
+            if result:
+                pkgs_with_vuln['result']['data'] += result['result']['data']
+
+        logger.info('%s took %0.2f secs for %s'
+                    'for total_results %d', self._request.external_request_id,
+                    time.time() - time_start, caller, len(pkgs_with_vuln['result']['data']))
+        return pkgs_with_vuln
+
+    def _filter_vulnerable_packages(self, vulnerabilities: List) -> Dict:
+        """Filter out vulnerabilities whose commit sha is out of vuln_commit_rules."""
+        logger.info('Executing filter_vulnerable_packages')
+
+        filter_vulnerabilities = defaultdict(list)
+        gh = GithubUtils()
+        for vuln in vulnerabilities:
+            package_name = vuln.get('package_name', [None])[0]
+            vuln_rules = vuln.get('vuln_commit_date_rules', [None])[0]
+            pseudo_version = self._normalized_packages.version_map.get(package_name)
+            if not pseudo_version:
+                logger.info("Not a Pseudo Version.")
+                continue
+            time_stamp = gh.extract_timestamp(pseudo_version)
+            if all([vuln_rules, time_stamp,
+                    gh._is_commit_date_in_vuln_range(time_stamp, vuln_rules)]):
+                filter_vulnerabilities[package_name].append(vuln)
+        return filter_vulnerabilities
+
+    def _get_golang_package_details(self, pkg_node) -> Tuple[Package, PackageDetails]:
+        """Get Pseudo Golang Package Details."""
+        pkg_name = pkg_node.get('name', [None])[0]
+        ecosystem = pkg_node.get('ecosystem', [''])[0]
+        pkg = Package(name=pkg_name, version=self._normalized_packages.version_map[pkg_name])
+        latest_version = pkg_node.get('latest_version', [''])[0]
+        public_vulns, private_vulns = self._get_vulnerabilities(
+            self.filtered_vul.get(pkg_name, []))
+        recommended_latest_version = pkg_node.get("latest_non_cve_version", [""])[0]
+        pkg_details = PackageDataWithVulnerabilities(
+            **pkg.dict(),
+            ecosystem=ecosystem,
+            latest_version=latest_version,
+            github={},
+            licenses=[],
+            url=_get_snyk_package_link(ecosystem, pkg_name),
+            private_vulnerabilities=private_vulns,
+            public_vulnerabilities=public_vulns,
+            recommended_version=recommended_latest_version)
+
+        return pkg, pkg_details
+
+    def _get_package_details_from_graph_for_pseudo_versions(self) -> List:
+        """Stack analyses call only for pseudo version applicable for golang."""
+        logger.info('Executing get_batch_sa_data_for_pseudo_version')
```

Comment:
nit: This should be a debug print because it already prints the elapsed time in the endof this function

Reply:
Ack!